### ISSUE TITLE
Insolation wiggles

### ISSUE THREAD

**Issue by ClimateTools** [brian-rose/climlab#128]
I plotted the summer insolation at 30N over the Holocene using the insolation function (v0.7.6) following the example in https://climlab.readthedocs.io/en/latest/api/climlab.solar.insolation.html, but there is a wiggle there. I'm not sure how it comes from. When I roll back to the version 0.7.0, the insolation returns normal without this weird wiggle.
![image](https://user-images.githubusercontent.com/7372813/93979051-758c0180-fd42-11ea-8c82-186d71daa49a.png)

**Comment by brian-rose** [comment]
Hi, can you please post the code that produces the figure above? Thanks.

**Comment by ClimateTools** [comment]
Yes, here is my code.

import numpy as np
import matplotlib.pyplot as plt
from climlab.solar.orbital import OrbitalTable
from climlab.solar.insolation import daily_insolation

years = np.linspace(-11, 0, 1001)

orb = OrbitalTable.interp(kyear=years)

S30 = daily_insolation(lat=30, day=172, orb=orb)

plt.plot(years*1000,S30)
plt.xlabel('Year')
plt.ylabel(r'Insolation (W/$m^{2}$)')

**Comment by brian-rose** [comment]
Thanks. I can see now that the problem is due to a discontinuity in the longitude of perihelion in the underlying data table:

```
print(OrbitalTable.long_peri.sel(kyear=[-6,-5]))
```
produces
```
<xarray.DataArray 'long_peri' (kyear: 2)>
array([539.99, 196.68])
Coordinates:
  * kyear    (kyear) int64 -6 -5
```

Prior to climlab 0.7.1 there was a `lookup_parameters()` method for the custom `OrbitalTable` object, which applied `np.unwrap()` to the `long_peri` data to deal with the 360º discontinuities. Somehow in the transition to the xarray version of `OrbitalTable` in 0.7.1 we missed this. 

We'll have to fix this. Thanks for bringing it up.

**Comment by brian-rose** [comment]
Here's a workaround:

```
import numpy as np
import matplotlib.pyplot as plt
import xarray as xr
from climlab.solar.orbital import OrbitalTable
from climlab.solar.insolation import daily_insolation

# apply np.unwrap to remove discontinuities in the longitude of perihelion
OrbitalTable['long_peri'] = np.rad2deg(xr.apply_ufunc(np.unwrap, np.deg2rad(OrbitalTable.long_peri)))

years = np.linspace(-11, 0, 1001)
orb = OrbitalTable.interp(kyear=years)
S30 = daily_insolation(lat=30, day=172, orb=orb)

plt.plot(years*1000,S30)
plt.xlabel('Year')
plt.ylabel(r'Insolation (W/$m^{2}$)')
```
produces this nice smooth figure:
![image](https://user-images.githubusercontent.com/9097307/94199893-83976a80-fe87-11ea-9ecb-a10d9868c840.png)

**Comment by brian-rose** [comment]
To fix the codebase, we just need to add a call to `np.unwrap` like that [around here](https://github.com/brian-rose/climlab/blob/12f0bb4269ee1a0e371472d3833663f87ed99c6b/climlab/solar/orbital/table.py#L29) where the data are first loaded into `OrbitalTable`

**Comment by ClimateTools** [comment]
Thanks, Brian! The problem is well solved by adding the np.unwrap function.

**Comment by spencerahill** [comment]
@brian-rose couldn't help but noticing that your insolation plot though nicely smooth has larger values, max ~515 W/m2, compared to @ClimateTools original, max ~480 W/m^2.  Did the discontinuity that caused the weird spiking also pull down the values on average?  (Also phasing is slightly shifted.)

**Comment by brian-rose** [comment]
Hi @spencerahill , I noticed that too! I think that @ClimateTools might have used different parameters in the first plot?

If I run this code:
```
import numpy as np
import matplotlib.pyplot as plt
from climlab.solar.orbital import OrbitalTable
from climlab.solar.insolation import daily_insolation

years = np.linspace(-11, 0, 1001)
orb = OrbitalTable.interp(kyear=years)
S30 = daily_insolation(lat=30, day=172, orb=orb)

plt.plot(years*1000,S30)
plt.xlabel('Year')
plt.ylabel(r'Insolation (W/$m^{2}$)')
```
I get this plot:
![image](https://user-images.githubusercontent.com/9097307/94302321-08db5780-ff3a-11ea-86b6-ebee3bab7d5a.png)
which clearly has the same bug, but has amplitude and phasing consistent with my fix above, rather than the original figure submitted by @ClimateTools .

**Comment by brian-rose** [comment]
Fixed in #129, closing this.

**Comment by brian-rose** [comment]
The fix is incorporated into the new version 0.7.7 release.
